pinot-spi/src/main/java/org/apache/pinot/spi/metrics/PinotMetricName.java

```diff
@@ -23,5 +23,20 @@
  */
 public interface PinotMetricName {
 
+  /**
+   * Returns the actual metric name.
+   */
   Object getMetricName();
+
+  /**
+   * Overrides the equals method. This is needed as {@link PinotMetricName} is used as the key of the key-value pair
+   * inside the hashmap in MetricsRegistry. Without overriding equals and hashCode method, all the existing k-v pairs
+   * stored in hashmap cannot be retrieved by initializing a new key.
+   */
+  boolean equals(Object obj);
+
+  /**
+   * Overrides the hashCode method.
+   */
+  int hashCode();
```

Comment:
I think that you also need to add the documentation here that `hashCode()`'s contract should be the same as `equals()`..

Reply:
updated.